test/_data/prefer_asserts_in_initializer_list/lib.dart

```diff
@@ -0,0 +1,7 @@
+class A {
+  A.c1(a) : assert(a != null); // OK
+  A.c2(a) { // LINT
+    assert(a != null);
+  }
+  A.c3(a) {} // OK
+}
```

Comment:
It would be good to have a test showing that a body with non-assert statements also isn't linted.